### PR TITLE
Extend AppError page to show special messages and actions when GETting a 404

### LINKED ISSUE #5666: Replace 'AppError' page with 'Page not found' when receiving a 404 response on a detail page
### Observed behavior

This issue is extracted from #5305. And is focused on this one case:

> Accessing a detail-type page when the underlying object no longer exists

Examples:

1. Viewing a User Permissions Page when the FacilityUser has been deleted
1. Viewing a Lesson Plan or Report page when the Lesson or Lesson Resource has been deleted
1. Visiting a Coach link for a class that has been deleted (it's handled like a 403 error)
1. Viewing ContentNode that has been deleted
1. Viewing a Quiz Plan or Report page when the Quiz has been deleted
1. Viewing a Learner Report page when the LEarner has been deleted
1. Visiting a report page for a learner or learner group when they have been deleted
1. (Add more)

(need to enumerate all cases to verify this issue is fixed)

In virtually all cases, if this happens, we get the generic AppError page.

### Expected behavior

When ever a user tries to *access* a detail-type page of deleted record, then they see a 'Page Not Found' screen that looks like this

Screen Shot 2019-06-13 at 2 59 45 PM

### User-facing consequences

Instead of seeing a generic AppError page, they will see a 404-like page that indicates that the thing they are trying to access has been deleted.

### Context

Kolibri Version 0.12.4

### PR DESCRIPTION
**Pull request by jonboiser**
### Summary

1. Adds a 404 page whenever the catch-all error handler `handleApiError` is used and the response shows it was a 404 resulting from a GET request.
1. ~Fix broken links on manage device permissions page~ (moved to #5976 )

Screen Shot 2019-08-12 at 2 08 41 PM
	

There are still a few places where the 404 page doesn't show because of something particular about how errors are handled. Here is the current list

1. Learn > Exam Report Viewer
1. Coach > most reports pages

### Reviewer guidance

You can get a 404 in most cases by modifying a UUID in a URL by a character. Make sure that the special 404 version of App Error shows up and that the default action button takes you to the root of the plugin you're in.

### References

Fixes #5666 

### Contributor Checklist

PR process:

- [x] PR has the correct target branch and milestone
- [x] PR has 'needs review' or 'work-in-progress' label
- [ ] If PR is ready for review, a reviewer has been added. (Don't use 'Assignees')
- [x] If this is an important user-facing change, PR or related issue has a 'changelog' label
- [ ] If this includes an internal dependency change, a link to the diff is provided

Testing:

- [ ] Contributor has fully tested the PR manually
- [ ] If there are any front-end changes, before/after screenshots are included
- [ ] Critical user journeys are covered by Gherkin stories
- [ ] Critical and brittle code paths are covered by unit tests

### Reviewer Checklist

- Automated test coverage is satisfactory
- PR is fully functional
- PR has been tested for [accessibility regressions](http://kolibri-dev.readthedocs.io/en/develop/manual_testing.html#accessibility-a11y-testing)
- External dependency files were updated if necessary (`yarn` and `pip`)
- Documentation is updated
- Contributor is in AUTHORS.md

### CODE REVIEW

**Comment by rtibbles** [comment]
I think just rebasing to current develop should resolve this.

File: kolibri/core/assets/src/views/AppError/index.vue [hunk 1]
```diff
@@ -51,9 +49,30 @@
...
   import { mapActions } from 'vuex';
   import ReportErrorModal from './ReportErrorModal';
 
+  // Inspects URL and returns an enum of the plugin (enum values are "learn", "coach", etc.)
+  function getCurrentKolibriPlugin() {
+    const url = window.location.href;
+
+    if (url.includes(urls['kolibri:learn:learn']())) {
+      return 'LEARN';
+    } else if (url.includes(urls['kolibri:coach:coach']())) {
```
Comment (jonboiser) [inline kolibri/core/assets/src/views/AppError/index.vue]: @rtibbles These URL names are probably wrong now. Can you share off the top of your head what they should be now?

File: kolibri/core/assets/src/views/AppError/index.vue [hunk 2]
```diff
@@ -51,9 +49,30 @@
...
+  import get from 'lodash/get';
+  import urls from 'kolibri.urls';
   import { mapActions } from 'vuex';
   import ReportErrorModal from './ReportErrorModal';
 
+  // Inspects URL and returns an enum of the plugin (enum values are "learn", "coach", etc.)
+  function getCurrentKolibriPlugin() {
+    const url = window.location.href;
+
+    if (url.includes(urls['kolibri:kolibri.plugins.learn:learn']())) {
```
Comment (rtibbles) [inline kolibri/core/assets/src/views/AppError/index.vue]: These checks will break if any of these plugins is disabled. Should always do a check that the URL exists:
`urls['kolibri:kolibri.plugins.learn:learn'] && url.includes(urls['kolibri:kolibri.plugins.learn:learn']())`

**Review by rtibbles — CHANGES_REQUESTED** [review]
Looks good, have not tested.

One small tweak needed.

**Review by rtibbles — APPROVED** [review]

File: kolibri/core/assets/src/views/AppError/index.vue [hunk 3]
```diff
@@ -83,12 +154,19 @@
...
       defaultErrorHeader: 'Sorry! Something went wrong!',
+      // eslint-disable-next-line
       defaultErrorExitPrompt: 'Back to home',
       pageReloadPrompt: 'Refresh',
       defaultErrorMessage:
         'We care about your experience on Kolibri and are working hard to fix this issue.',
       defaultErrorResolution: 'Try refreshing this page or going back to the home page.',
       defaultErrorReportPrompt: 'Help us by reporting this error',
+      pageNotFoundHeader: 'Page not found',
+      pageNotFoundMessage: "Sorry, we can't seem to find the page you're looking for.",
```
Comment (indirectlylit) [inline kolibri/core/assets/src/views/AppError/index.vue]: unnecessary trailing period. From the [writing guidelines](http://kolibribeta.learningequality.org/design/patterns/writing):

> Skip periods and unnecessary punctuation

**Review by indirectlylit — COMMENTED** [review]
awesome!

File: kolibri/core/assets/src/views/AppError/index.vue (hunk 3 above)
Comment (indirectlylit) [reply kolibri/core/assets/src/views/AppError/index.vue]: Actually, looks like we forgot to incorporate the latest strings: https://docs.google.com/document/d/1B7E9RrxLatDF8R9a8JG8W8tEez6KrjrwPQPbE1biKsg/edit#

**Pull request merged**